src/fontra/views/editor/editor.js

```diff
@@ -2818,6 +2844,24 @@ export class EditorController {
     this.sceneSettings.selectedSourceIndex = newSourceIndex;
   }
 
+  async doSelectPreviousNextGlyph(selectPrevious) {
+    const panel = this.getSidebarPanel("glyph-search");
+    const glyphNames = panel.glyphsSearch.getFilteredGlyphNames();
+    if (!glyphNames.length) {
+      return;
+    }
+
+    const selectedGlyphName = this.sceneSettings.selectedGlyphName;
+    if (!selectedGlyphName) {
+      return;
+    }
+    const index = glyphNames.indexOf(selectedGlyphName);
```

Comment:
It is possible that the selected glyph isn’t in the list. We may want to fall back to the last or first item in the list, depending on prev or next.

Reply:
@justvanrossum If `selectedGlyphName` is not part of `glyphNames`, `.indexOf()` returns -1. And then `newIndex` becomes 0. So, it falls back to the first item of the list already. Or is there something I have overlooked?

Reply:
It will be 0 for "next glyph", but it will be -2 for "previous glyph", which makes it select the glyph _before_ the last glyph. So this is totally accidental. The fact that it "doesn't crash" isn't good enough.

Suggest:
```js
    const newIndex =
      index == -1
        ? selectPrevious
          ? glyphNames.length - 1
          : 0
        : modulo(index + (selectPrevious ? -1 : 1), glyphNames.length);
```

Please test by putting a glyph into edit mode, then filter the search list so it doesn't contain that glyph.

Reply:
Ah sorry for another loop. I have only checked 'next glyph' by value and missed to look at the value of the previous glyph. I changed the code as suggested and tested it: previous AND next. It works well now.